### DISCUSSION TITLE
Possibility to create groups of fields

### DISCUSSION THREAD

**Discussion by scoch5** [craftcms/cms#14600] [Ideas] [answered]
It would be very useful to be able to group multiple fields into one "group" field which allows them to be added in bulk.
A use case could be one in which there are a series of settings to be used multiple times for example in matrix blocks (e.g. padding, spacing, bg color, etc..).

Thanks

**Comment by terryupton** [comment]
I have come across this today when working on a new Craft 5 project. 
I added my comments to this issue:
https://github.com/craftcms/cms/discussions/14015

**Comment by bencarr** [comment]
I really like this idea! I'd love if the groups specifically supported:
- **UI elements**, so my group could have a heading or a tip/warning
- **Widths**, so I can specify some of the group should be side-by-side (I think I'd be ok if this meant groups always had to be full-width when they're placed in a layout)
- **Conditions**, so fields in the group could be dependent on one another, or apply a user condition
- **Detaching**, so I could ungroup the fields in a specific entry type and make instance-level changes if I need to

A few content modeling examples I've run into that would benefit from this:
- **SEO fields** — Like Title, Description, Image, and Robots. Typically added to an "SEO" tab on every individually-viewable entry type
- **Hero fields** — Like Image, Heading, Subtitle, Description, CTA. Typically added at the top of the "Content" tab on several entry types, with varying other fields below. Sometimes I'll use a "Hero Style" field to controls the visibility of some of the other fields, like maybe one style is a solid color instead of an image, so it'll hide the image field and show a color picker.
- **Date ranges** — Sometimes I'll have a few entry types that have Start Date and End Date fields, which would be nice to keep together.
- **Shared taxonomy fields** — Sometimes multiple entry types relate to the same set of taxonomies, so I'll have a few Entries fields that I add to several entry types

**Reply by thanasispsarros** [reply]
All the features you are describing (except for detaching, but this can be solved with other ways) are already available in entry types. Since matrix blocks are now entry types, we only need ["static" matrix blocks](https://github.com/craftcms/cms/discussions/14015) to have a field group with the above features.

Example:

- I define an entry type with all the fields I want to have in a group.
- I define a matrix field with only this entry type and min/max 1 block.
-  This field should be _static_, meaning the author can't add blocks, remove blocks, or disable blocks. (This is the missing functionality).

The reason I'm going about it this way is that IMO there is little reason to introduce a new concept when the funcionality is 95% ready with what we have now. Of course there may be technical reasons to not do it via a Matrix field, not sure about that.

**Reply by alexanderbuergin** [reply]
What if we could define reusable Field Group Types, similar to Entry Types?

You’d create one group — e.g. «SEO Fields» with Title, Description, Image, Robots — and reuse it across entry types or matrix blocks. Same for common setups like «Hero Section» or «Date Range».

This would avoid repeating the same set of fields over and over, improve maintainability, and keep things consistent across the project.

Think of it as modular field modeling: define once, reuse anywhere.

**Comment by philipschilling** [comment]
It would help me if I can assign fields to a group and thereby better distinguish the fields since I created many fields for different datasets.

**Comment by croxton** [comment]
In addition to fields, I would also like to be able to group entry types which I have found can quickly get out of hand.

**Reply by willbocc** [reply]
Something like the old field groups would be ideal here.

**Reply by thanasispsarros** [reply]
Indeed, but this discussion is not about grouping fields in the field index, but in the field layout.

**Reply by terryupton** [reply]
@croxton - I added a similar comment about this here on another thread.
https://github.com/craftcms/cms/discussions/14726#discussioncomment-8985677

**Reply by brettburwell** [reply]
🙏 Please please please bring this back. Seems odd that they did away with it.

**Reply by brandonkelly** [reply]
@brettburwell This FR is not asking to bring back the former concept of field groups. That was explained here: #14083

**Reply by brettburwell** [reply]
@brandonkelly Thanks for the reply, I'll check that thread out. But I'd still give an incredibly enthusiastic +1 for @croxton's suggestion of adding support for grouping/organizing entry types (if we should start that idea as a stand alone FR let me know). 

With sites of any size / complexity, like Mark mentioned, it quickly gets a little chaotic feeling. 

I've seen folks suggest using a naming convention to self sort the entry types (e.g. `Block - Image`). That help organize things on the DX side of things, but because that same label shows up for authors (assuming it's a matrix block) in the matrix block's heading bar on an entry page, it's a major drawback on the AX side of things.

Having a simple system to keep things more organized would be super useful.

**Comment by richhayler** [comment]
In my opinion it would be even better if you could define a field group and then add fields to it - i.e. fields could be a member of more than one group. Because fields can be used more flexibly in Craft 5 this feature would provide more flexibility for organising this information.

**Comment by ryanleichty** [comment]
Had the same idea coming from Sanity with their `object` type schema that allows nested fields: https://github.com/craftcms/cms/discussions/15626

![CleanShot 2024-08-29 at 09 52 12@2x](https://github.com/user-attachments/assets/da50186d-6c62-404c-83ec-281a0c516b3b)

**Comment by kobeaerts** [comment]
Any updates on this one? 
This is the most upvoted issue of the past year, would be cool if we can get this in 2025! 🔥

**Comment by alexanderbuergin** [comment]
What if we could define reusable Field Group Types, similar to Entry Types?

You’d create one group — e.g. «SEO Fields» with Title, Description, Image, Robots — and reuse it across entry types or matrix blocks. Same for common setups like «Hero Section» or «Date Range».

This would avoid repeating the same set of fields over and over, improve maintainability, and keep things consistent across the project.

Think of it as modular field modeling: define once, reuse anywhere.

**Comment by thupsi** [comment]
Related: https://github.com/wsydney76/craft-properties-field

**Reply by leevigraham** [reply]
Just saw this in the discord and immediately remembered this discussion.

**Comment by FalkoJoseph** [comment]
Following this. Right now I'm using a nested matrix.

**Comment by ryanleichty** [comment]
I recently noticed this fieldset layout in the field settings menu. The concept of a fieldset makes sense at least from a UI perspective.

![CleanShot 2025-06-12 at 10 14 19@2x](https://github.com/user-attachments/assets/f6e31b3e-eaa7-41ac-b94c-cd0596be55e7)

**Comment by leevigraham** [comment]
@brandonkelly is working on this: https://github.com/craftcms/cms/pull/17424

**Reply by ryanleichty** [reply]
Awesome! 👏

**Reply by FalkoJoseph** [reply]
> @brandonkelly is working on this: #17424

Very cool! I moved to Craft from WordPress at just the right moment 😄

**Comment by brandonkelly** [comment, marked as answer]
Craft 5.8 is out, which introduces a new “Content Block” field type, which can be used to create reusable nested field layouts. (#17424)
